### ISSUE TITLE
[Bug Report] Crash when using device discovery in 1.7.2

### ISSUE THREAD

**Issue by cweiske** [Domi04151309/HomeApp#29]
**Description**

The experimental device discovery in HomeApp 1.7.2 crashes here in my home network.

I'm pretty sure it has to do with one of the devices here in the network, but I don't know which.

**Logs**
```
I WifiService: acquireMulticastLock uid=10088
I mi04151309.hom: Background concurrent copying GC freed 28261(3249KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 5807KB/11MB, paused 2.376ms total 284.427ms
D Volley  : [1663] e.c: /data/user/0/io.github.domi04151309.home/cache/volley/1061423297142620875: java.io.FileNotFoundException: /data/user/0/io.github.domi04151309.home/cache/volley/1061423297142620875: open failed: ENOENT (No such file or directory)
D Volley  : [1663] e.o: Could not delete cache entry for key=http://192.168.3.3:49152/description.xml, filename=1061423297142620875
D Volley  : [1658] e.c: /data/user/0/io.github.domi04151309.home/cache/volley/1061423297142620875: java.io.FileNotFoundException: /data/user/0/io.github.domi04151309.home/cache/volley/1061423297142620875: open failed: ENOENT (No such file or directory)
D Volley  : [1658] e.o: Could not delete cache entry for key=http://192.168.3.3:49152/description.xml, filename=1061423297142620875
I mi04151309.hom: Background young concurrent copying GC freed 80112(8928KB) AllocSpace objects, 12(240KB) LOS objects, 50% free, 5748KB/11MB, paused 680us total 106.160ms
I WifiService: releaseMulticastLock uid=10088
D AndroidRuntime: Shutting down VM
E AndroidRuntime: FATAL EXCEPTION: main
E AndroidRuntime: Process: io.github.domi04151309.home, PID: 1093
E AndroidRuntime: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:501)
E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)
E AndroidRuntime: Caused by: java.lang.reflect.InvocationTargetException
E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
E AndroidRuntime: 	... 1 more
E AndroidRuntime: Caused by: org.xmlpull.v1.XmlPullParserException: expected: /link read: head (position:END_TAG </head>@18:9 in java.io.InputStreamReader@2c10d77) 
E AndroidRuntime: 	at com.android.org.kxml2.io.KXmlParser.readEndTag(KXmlParser.java:984)
E AndroidRuntime: 	at com.android.org.kxml2.io.KXmlParser.next(KXmlParser.java:375)
E AndroidRuntime: 	at com.android.org.kxml2.io.KXmlParser.next(KXmlParser.java:313)
E AndroidRuntime: 	at b.a.a.a.h(Unknown Source:244)
E AndroidRuntime: 	at b.a.a.a.g(Unknown Source:5)
E AndroidRuntime: 	at b.a.a.b$c.b(Unknown Source:7)
E AndroidRuntime: 	at b.a.a.b$c.a(Unknown Source:2)
E AndroidRuntime: 	at b.b.a.w.n.O(Unknown Source:8)
E AndroidRuntime: 	at b.b.a.w.n.e(Unknown Source:2)
E AndroidRuntime: 	at b.b.a.f$b.run(Unknown Source:30)
E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7356)
E AndroidRuntime: 	... 3 more
W ActivityTaskManager:   Force finishing activity io.github.domi04151309.home/.activities.SearchDevicesActivity
I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
I Process : Sending signal. PID: 1093 SIG: 9
I WindowManager: WIN DEATH: Window{6ced571 u0 io.github.domi04151309.home/io.github.domi04151309.home.activities.SearchDevicesActivity}
I ActivityManager: Process io.github.domi04151309.home (pid 1093) has died: fore TOP 
I WindowManager: WIN DEATH: Window{ec0885f u0 io.github.domi04151309.home/io.github.domi04151309.home.activities.DevicesActivity}
I Zygote  : Process 1093 exited due to signal 9 (Killed)
```

**Additional information**

 - Android version: 10
 - Home App version: 1.7.2
 - Installation source: F-Droid

**Comment by cweiske** [comment]
This has probably been fixed by commit 87c9ad785ea57a8070503370796caf6e05d20bb9 which has not been released yet.

**Comment by cweiske** [comment]
I'll wait for a new release.
